packages/@react-spectrum/color/docs/ColorArea.mdx

```diff
@@ -0,0 +1,310 @@
+    defaultValue="hsl(0, 100%, 50%)"
+    xChannel="saturation"
+    yChannel="lightness" />
+  <div>
+    <Label
+      id="hsl-aria-labelledby-id">Background color</Label>
+    <ColorArea
+      aria-labelledby="hsl-aria-labelledby-id"
+      defaultValue="hsl(0, 100%, 50%)"
+      xChannel="saturation"
+      yChannel="lightness" />
+  </div>
+</Flex>
+```
+
+### Accessibility
+
+#### Role description
+
+In order to communicate to a screen reader user that the ColorArea adjusts in two-dimensions,
+ColorArea provides an `aria-roledescription`, using the localized string "2D Slider", on each of the
+visually hidden `<input>` elements.
+
+#### Value formatting
+
+The `aria-valuetext` of each `<input>` element within the ColorArea is formatted according to the user's locale automatically.
+It will include the localized color channel name and current value for each channel, with the channel name
+and value that the `<input>` element controls coming before the channel name and value for the adjacent
+`<input>` element. For example, for an RGB ColorArea where the `xChannel` is "blue", the `yChannel`
+is "green", when the current selected color is yellow (`rgb(255, 255, 0)`), the `<input>` representing the
+blue channel will have `aria-valuetext` to announce as "Blue: 0, Green: 255", and the `<input>`
+representing the green channel will have `aria-valuetext` to announce as "Green: 255, Blue: 0".
+
+### Internationalization
+
+For languages that are read right-to-left (e.g. Hebrew and Arabic), the layout of the ColorArea is automatically flipped. As mentioned previously,
+ColorArea automatically uses the current locale to format the `aria-valuetext` for each of the `<input>` element with the channel name and current value.
+
+## Events
+
+ColorArea supports two events: `onChange` and `onChangeEnd`. `onChange` is triggered whenever the ColorArea's handle is dragged, and `onChangeEnd`
+is triggered when the user stops dragging the handle. Both events receive a <TypeLink links={colorTypes.links} type={colorTypes.exports.Color} /> object
+as a parameter.
+
+The example below uses `onChange` and `onChangeEnd` to update two separate elements with the ColorArea's value.
+
+```tsx example
+function Example() {
+  let [currentValue, setCurrentValue] = React.useState(parseColor('hsl(50, 100%, 50%)'));
+  let [finalValue, setFinalValue] = React.useState(parseColor('hsl(50, 100%, 50%)'));
+
+  return (
+    <div>
+      <ColorArea
+        value={currentValue}
+        onChange={setCurrentValue}
+        onChangeEnd={setFinalValue} />
+      <pre>Current value: {currentValue.toString('hsl')}</pre>
+      <pre>Final value: {finalValue.toString('hsl')}</pre>
+    </div>
+  );
+}
+```
+
+## Creating a color picker
+
+To build a fully functional color picker, the ColorArea control, that adjusts two channels of a color value stored in state,
+should be combined with a separate control, like a [ColorSlider](ColorSlider.html) or [ColorWheel](useColorWheel.html), for
```

Comment:
Maybe re-word this a bit. Not sure if this would be much better:

```suggestion
To build a fully functional color picker, combine a ColorArea, which adjusts two channels of a color value stored in state, with a separate control, like a [ColorSlider](ColorSlider.html) or [ColorWheel](useColorWheel.html), for
```